File: 01-Programming-Languages/Dart/Algorithms/Arrays/two_sum.dart
List<int> twoSum(List<int> nums, int target) {
  final numMap = <int, int>{};

  for (var i = 0; i < nums.length; i++) {
    final complement = target - nums[i];
    if (numMap.containsKey(complement)) {
      return [numMap[complement]!, i];
    }
    numMap[nums[i]] = i;
  }

  return [];
}

void main() {
  final nums = [2, 7, 11, 15];
  final target = 9;
  print('Indices: ${twoSum(nums, target)}');
}

File: 01-Programming-Languages/Kotlin/Algorithms/Arrays/two_sum.kt
fun twoSum(nums: IntArray, target: Int): Pair<Int, Int>? {
    val numMap = mutableMapOf<Int, Int>()

    for ((i, num) in nums.withIndex()) {
        val complement = target - num
        numMap[complement]?.let { index ->
            return Pair(index, i)
        }
        numMap[num] = i
    }

    return null
}

fun main() {
    val nums = intArrayOf(2, 7, 11, 15)
    val target = 9

    when (val result = twoSum(nums, target)) {
        is Pair -> println("Indices: (${result.first}, ${result.second})")
        else -> println("No solution found")
    }
}